bin/interactive.py

```diff
@@ -305,15 +282,16 @@ def kill_submission_handler(signal, frame):
             # Possibly already written by the alarm.
             if submission_time is None:
                 submission_time = rusage.ru_utime + rusage.ru_stime
-            left -= 1
-            first_done = False
-            continue
-
-        if interaction:
+        elif interaction:
             if pid == team_tee_pid or pid == val_tee_pid:
-                left -= 1
-                first_done = False
-                continue
+                pass
+            else:
+                assert False
+        else:
+            assert False
+
+        left -= 1
+        first_done = False
```

Comment:
hrmmm, I think `first_done` should only become false for submission and validator finishing. If the validator closes stdout, that should kill one of the `tee`s, but by that time the validator can still finish before the submission after which we want to kill the submission. Maybe we should just introduce separate `done` indicators for each of the non-validator processes and kill the ones that are not done yet?

Reply:
a tee should never end first but i changed it anyway. We don't need extra state, we can kill groups :D